hoomd/md/EvaluatorPairMie.h

```diff
@@ -141,16 +149,16 @@ class EvaluatorPairMie
             if (rsq < rcutsq && mie1 != 0)
                 {
                 Scalar r2inv = Scalar(1.0)/rsq;
-                Scalar rninv = pow(r2inv,mie3/Scalar(2.0));
-                Scalar rminv = pow(r2inv,mie4/Scalar(2.0));
+                Scalar rninv = fast::pow(r2inv,mie3/Scalar(2.0));
+                Scalar rminv = fast::pow(r2inv,mie4/Scalar(2.0));
                 force_divr= r2inv * (mie3 * mie1 * rninv - mie4 * mie2 * rminv);
 
                 pair_eng = mie1 * rninv - mie2 * rminv;
 
                 if (energy_shift)
                     {
-                    Scalar rcutninv = Scalar(1.0)/pow(rcutsq,mie3/Scalar(2.0));
-                    Scalar rcutminv = Scalar(1.0)/pow(rcutsq,mie4/Scalar(2.0));
+                    Scalar rcutninv = Scalar(1.0)/fast::pow(rcutsq,mie3/Scalar(2.0));
+                    Scalar rcutminv = Scalar(1.0)/fast::pow(rcutsq,mie4/Scalar(2.0));
```

Comment:
```suggestion
                    Scalar rcutninv = fast::pow(rcutsq,-mie3/Scalar(2.0));
                    Scalar rcutminv = fast::pow(rcutsq,-mie4/Scalar(2.0));
```
You can save a few cycles using a negative exponent in place of division.

You might try the same to avoid the 1.0/rsq above and see if it makes any measurable change in performance.

Reply:
I tested the performance both `1/rsq` and the negative exponent in the non-energy shifted calculation, and `1/rsq` performs marginally better, presumably because `r2inv` is calculated and stored for the following force calculation anyway.